client/dashboard/sites/deployments-list/trigger-deployment-modal-form.tsx

```diff
@@ -116,14 +117,40 @@ export function TriggerDeploymentModalForm( {
 		);
 	};
 
+	const getDeploymentMessage = () => {
+		if ( ! selectedDeployment ) {
+			return __(
+				"You're about to deploy changes from the selected repository to your production site. This will replace contents of your live site with the selected repository."
+			);
+		}
+
+		const deploymentType = getDeploymentTypeFromPath( selectedDeployment.target_dir );
+
+		switch ( deploymentType ) {
+			case 'plugin':
+				return __(
+					"You're about to deploy changes from the selected repository to your production site. This will replace the plugin in your selected deployment location. The rest of your site will remain unchanged."
+				);
+			case 'theme':
+				return __(
+					"You're about to deploy changes from the selected repository to your production site. This will replace the theme in your selected deployment location. The rest of your site will remain unchanged."
+				);
```

Comment:
Nitpick: These two sentences are exactly the same except for the theme and plugin. Can we combine them into one string? 
```suggestion
			case 'plugin':
			case 'theme':
				return sprintf( 
  				     __(
  					       "You're about to deploy changes from the selected repository to your production site. This will replace the %(type)s in your selected deployment location. The rest of your site will remain unchanged.", {type: deploymentType }
  				       )
				);
```

Reply:
I would vote for keeping the strings separate. While merging them indeed simplifies the code, it can also make it difficult for translators in certain locales where they need more flexibility - e.g. put the `plugin` or `theme` word elsewhere in the string (or if the words have different genders).